operations/host_spawn.go

```diff
@@ -1375,6 +1375,61 @@ Examples:
 	}
 }
 
+func hostFindWithIpAddress() cli.Command {
+	const (
+		ipAddressFlagName = "ip-address"
+		spruceFlagName    = "spruce"
+	)
+	return cli.Command{
+		Name:  "find-by-ip-address",
+		Usage: "find host using ip address",
+		Flags: addHostFlag(
+			cli.StringFlag{
+				Name:  joinFlagNames(ipAddressFlagName, "ip"),
+				Usage: "ip address used to find host",
+			},
+			cli.BoolFlag{
+				Name:  spruceFlagName,
+				Usage: "returns host link in spruce",
+			},
+		),
+		Action: func(c *cli.Context) error {
+			confPath := c.Parent().Parent().String(confFlagName)
+			ipAddress := c.String(ipAddressFlagName)
+			spruce := c.Bool(spruceFlagName)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			conf, err := NewClientSettings(confPath)
+			if err != nil {
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			client := conf.getRestCommunicator(ctx)
+			defer client.Close()
+
+			host, err := client.FindHostByIpAddress(ctx, ipAddress)
+			if err != nil {
+				return errors.Wrapf(err, "failed to fetch host with ip address '%s'", ipAddress)
+			}
+			if host == nil {
+				return errors.New(fmt.Sprintf("Host with ip address '%s' not found", ipAddress))
+			}
+
+			link := fmt.Sprintf("https://evergreen.mongodb.com/host/%s", *host.Id)
```

Comment:
Rather than referencing the pointers directly, it's best to use utility.FromStringPtr to avoid a nil pointer error